controllers/idler/idler_controller.go

```diff
@@ -141,6 +155,76 @@ func (r *Reconciler) ensureIdling(logger logr.Logger, idler *toolchainv1alpha1.I
 	return r.updateStatusPods(idler, newStatusPods)
 }
 
+func (r *Reconciler) createNotification(logger logr.Logger, idler *toolchainv1alpha1.Idler) (bool, error) {
+	//Get the HostClient
+	hostCluster, ok := r.GetHostCluster()
+	if !ok {
+		return false, fmt.Errorf("unable to get the host cluster")
+	}
+	// add sending notification here
+	//check the condition on Idler if notification already sent
+	// only create a notification if not created before
+	_, found := condition.FindConditionByType(idler.Status.Conditions, toolchainv1alpha1.IdlerTriggeredNotificationCreated)
+	if !found || condition.IsFalse(idler.Status.Conditions, toolchainv1alpha1.IdlerTriggeredNotificationCreated) {
+		userEmails, err := r.getUserEmailFromMUR(logger, hostCluster, idler)
+		if err != nil {
+			return false, err
+		}
+		if len(userEmails) > 0 {
+			for _, userEmail := range userEmails {
+				_, err := notify.NewNotificationBuilder(hostCluster.Client, hostCluster.OperatorNamespace).
```

Comment:
To answer your question - if the notification was already but the status wasn't set, in the next reconcile it would create the notification again and try setting the status. This is because the status update is the only way currently that it is keeping track of whether a notification was sent, so a false condition would imply notification wasn't created even if it was.
I can improve message in condition such that the creation of notification looks at the message and reason of the condition as well, not just the status.
But, to confirm, isn't the situation when the Idler status is unable to update a rare case? Like Unable to Get a resource?